cmd/clusterctl/internal/util/image.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// ImageSplit contains all the information resulting from a split of an image name into its composing parts.
+type ImageSplit struct {
+	// Repository where the image is hosted
+	Repository string
+
+	// Name of the image
+	Name string
+
+	// Tag of the image
+	Tag string
+}
+
+// NewImageSplit returns a ImageSplit starting from an image name.
+func NewImageSplit(image string) (*ImageSplit, error) {
+	ret := &ImageSplit{}
+
+	imageSplit := strings.Split(image, ":")
+	repositoryAndImage := imageSplit[0]
```

Comment:
possibly better to check for `!= 2` here and return an early error.
if `len(imageSplit) == 1` the logic will reach:
`imageNameIndex := strings.LastIndex(repositoryAndImage, "/")`

Reply:
it accepts an image with onliny name (like e.g. "nginx"), si it is fine to move on